src/rules.c

```diff
@@ -692,12 +693,13 @@ static int check_subject(lnode *r, event_t *e)
 				//  match.  Any child dir would also match.
 				if (type == EXE_DIR) {
 					int rc = subj_dir_test(&(r->s[cnt]),
-								subj);
+								subj,
+								is_subj_trusted(e));
 					if (rc == 0)
 						return 0;
 				} else if (type == EXE &&
-				   strcmp(r->s[cnt].str, "untrusted")==0) {
-					if (check_trust_database(subj->str))
+					strcmp(r->s[cnt].str, "untrusted")==0) {
+						if (is_subj_trusted(e))
 						return 0;
```

Comment:
Please fix indentation for this branch and proper spacing between equivalence operands (might as well do it while you're here).